### DISCUSSION TITLE
"The argument of type '0x${string}' is not assignable to the parameter of type 'readonly [0x${string}]'."

### DISCUSSION THREAD

**Discussion by Stachelabs** [wevm/viem#1480] [Question]
Hi everyone,

I face this issue a lot when using Typescript with viem:

"The argument of type '0x${string}' is not assignable to the parameter of type 'readonly [0x${string}]'."

here is the code for example:

```
const [wallet] = await L2Wallets.getAddresses();
const balance = await tokenContract.read.balanceOf(wallet);
```

this error refers to the above read method when using wallet as a parameter.

Also,

I had same issue when using a contract address in getContract method:

```
const tokenContract = getContract({
  address: contractAddress as `0x${string}`,
  abi: ERC20ABI,
  publicClient,
});
```

I had to use 

```
as `0x${string}`

```

but when I past the raw contract from the variable, there is not problem.

Can someone help?

thank you very much

**Comment by emretepedev** [comment]
you should pass parameters as array:

```ts
const balance = await tokenContract.read.balanceOf([wallet]);
```

**Comment by eik-1** [comment]
I hate typescript
